src/Symfony/Component/PropertyAccess/PropertyAccessor.php

```diff
@@ -154,15 +163,17 @@ class PropertyAccessor implements PropertyAccessorInterface
      * Should not be used by application code. Use
      * {@link PropertyAccess::createPropertyAccessor()} instead.
      *
-     * @param bool                   $magicCall
-     * @param bool                   $throwExceptionOnInvalidIndex
-     * @param CacheItemPoolInterface $cacheItemPool
+     * @param bool                          $magicCall
+     * @param bool                          $throwExceptionOnInvalidIndex
+     * @param CacheItemPoolInterface        $cacheItemPool
+     * @param ClassMetadataFactoryInterface $classMetadataFactory
      */
-    public function __construct($magicCall = false, $throwExceptionOnInvalidIndex = false, CacheItemPoolInterface $cacheItemPool = null)
+    public function __construct($magicCall = false, $throwExceptionOnInvalidIndex = false, CacheItemPoolInterface $cacheItemPool = null, MetadataFactoryInterface $classMetadataFactory = null)
     {
         $this->magicCall = $magicCall;
         $this->ignoreInvalidIndices = !$throwExceptionOnInvalidIndex;
         $this->cacheItemPool = $cacheItemPool instanceof NullAdapter ? null : $cacheItemPool; // Replace the NullAdapter by the null value
+        $this->classMetadataFactory = $classMetadataFactory;
```

Comment:
What about instantiating it automatically (with minimal support) when not provided ?